README.md

```diff
@@ -179,6 +179,151 @@ At a glance you can see that the activations of the middle hidden node
+
+Below is an example using the XOR function: (assume we already had a keras model; you can get
+one using the codes above)
+
+```python
+    from networkx import set_node_attributes, set_edge_attributes
+    
+    # "model" is already a trained keras model.
+    dg = DenseGraph(model)
+    
+    # Get the networkx graph.
+    the_graph = dg.get_graph()
+    
+    # Loop through the graph by looping through the model.
+    for l in range(len(model.layers)):  # get number of layers
+        layer = model.layers[l]
+        for n in range(0, layer.input_shape[1]):  # get number of nodes in that layer
+            set_node_attributes(the_graph, {
+                str(l) + str(n): {  # "str(l) + str(n)" is the index of the node
+                    'shape': "diamond",
+                    'color': "#00ff00",
+                    'label': ""
+                }
+            })
+    
+            # Here, set the attributes for nodes in the output layer, which is the last
+            # layer of this model.
+            for h in range(0, layer.output_shape[1]):
+                if l == len(model.layers) - 1:  # check if the layer is the last layer
+                    set_node_attributes(the_graph, {
+                        str(l + 1) + str(h): {
+                            'shape': "square",
+                            'color': "#ff0000",
+                            'label': ""
+                        }
+                    })
+                # Now set the attributes of edges.
+                # The index of an edge is an tuple consisting of the indexes of 
+                # the 2 nodes that are connected by it.
+                set_edge_attributes(the_graph, {
+                    (str(l) + str(n), str(l + 1) + str(h)): {
+                        'color': "#0000ff"
+                    }
+                })
+    
+    # Pass the modified network graph to DenseGraph. 
+    dg.set_graph(the_graph)
+    
+    # Get the visualization & set the color and marker in the pyplot graph.
+    dg.render(X, x_color="#FF0000", x_marker="^")
+```
+
+Of course, you do not need to loop through the model if you already know the
+number of layers and number of nodes in each layer. You can directly set the 
+attributes of every node by its index using `set_node_attributes`. 
+
+The index of each node is defined as 
+a string, which equals the index of the layer (starts from 0) plus the index of
+the node in that layer (starts from 0). For example, the index "00" means the 
+first node ("0") in the first layer ("0").
+
+Similarly, you can use `set_edge_attributes` to directly set the attributes of 
+an edge, given you know its index. The index of an edge is a tuple consisting 
+of 2 nodes that are connected by it. The node in the upper layer usually comes 
+first. For example, ("00", "10") is the index of an edge that connects node 
+"00" and node "10".
+
+Below is the code example. It will give the same result as the codes above.
+
+```python
```

Comment:
Now that I see the code I am convinced the indexing is wrong / ugly and I agree with your original approach. Let's keep this for now as this is the current behavior - you can change this README when you fix https://github.com/gallettilance/kviz/issues/10